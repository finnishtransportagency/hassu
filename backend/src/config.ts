import { log } from "./logger";
import * as AWSXRay from "aws-xray-sdk";

const config = {
  projektiTableName: process.env.TABLE_PROJEKTI,
  projektiArchiveTableName: process.env.TABLE_PROJEKTI_ARCHIVE,
  cognitoURL: process.env.COGNITO_URL,
  velhoAuthURL: process.env.VELHO_AUTH_URL,
  velhoApiURL: process.env.VELHO_API_URL,
  velhoUsername: process.env.VELHO_USERNAME,
  velhoPassword: process.env.VELHO_PASSWORD,

  personSearchApiURL: process.env.PERSON_SEARCH_API_URL,
  personSearchApiURLProd: process.env.PERSON_SEARCH_API_URL_PROD,
  personSearchUsername: process.env.PERSON_SEARCH_API_USERNAME,
  personSearchPassword: process.env.PERSON_SEARCH_API_PASSWORD,
  personSearchAccountTypes: process.env.PERSON_SEARCH_API_ACCOUNT_TYPES?.split(","),
  personSearchUpdaterLambdaArn: process.env.PERSON_SEARCH_UPDATER_LAMBDA_ARN || "",

  searchDomain: process.env.SEARCH_DOMAIN,

  env: process.env.ENVIRONMENT,

  frontendDomainName: process.env.FRONTEND_DOMAIN_NAME,

  frontendPrivateKey: process.env.FRONTEND_PRIVATEKEY,

  uploadBucketName: process.env.UPLOAD_BUCKET_NAME,
  yllapitoBucketName: process.env.YLLAPITO_BUCKET_NAME,
  publicBucketName: process.env.PUBLIC_BUCKET_NAME,
  internalBucketName: process.env.INTERNAL_BUCKET_NAME,
  archiveBucketName: process.env.ARCHIVE_BUCKET_NAME,

  smtpKeyId: process.env.SMTP_KEY_ID,
  smtpSecret: process.env.SMTP_SECRET,

  emailsOn: process.env.EMAILS_ON,
  emailsTo: process.env.EMAILS_TO,
  isProd: (): boolean => process.env.ENVIRONMENT == "prod",
};

process.env.AWS_XRAY_CONTEXT_MISSING = "LOG_ERROR";
AWSXRay.setContextMissingStrategy("IGNORE_ERROR");

// Create your own logger, or instantiate one using a library.
AWSXRay.setLogger({
  error: (message, meta) => {
    log.error(message, { meta });
  },
  warn: (message, meta) => {
    log.warn(message, { meta });
  },
  info: (message, meta) => {
    // Reduce unnecessary logging in local development
    if (process.env.NODE_ENV !== "development") {
      log.info(message, { meta });
    }
  },
  debug: () => {
    // Ignore
  },
});

export { config };
