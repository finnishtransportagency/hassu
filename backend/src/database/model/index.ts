export * from "./common";
export * from "./projekti";
export * from "./suunnitteluVaihe";
export * from "./nahtavillaoloVaihe";
export * from "./hyvaksymisPaatosVaihe";
