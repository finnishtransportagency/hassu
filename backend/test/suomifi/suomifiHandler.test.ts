import { SQSEvent } from "aws-lambda";
import { setLogContextOid } from "../../src/logger";
import { SuomiFiSanoma, handleEvent, setMockSuomiFiClient } from "../../src/suomifi/suomifiHandler";
import { identifyMockUser } from "../../src/user/userService";
import { mockClient } from "aws-sdk-client-mock";
import { DynamoDBDocumentClient, GetCommand, UpdateCommand } from "@aws-sdk/lib-dynamodb";
import { parameters } from "../../src/aws/parameters";
import * as sinon from "sinon";
import { config } from "../../src/config";
import { DBMuistuttaja } from "../../src/muistutus/muistutusHandler";
import { now } from "lodash";
import { emailClient } from "../../src/email/email";
import { assert, expect } from "chai";
import { SuunnittelustaVastaavaViranomainen } from "hassu-common/graphql/apiModel";
import { PdfViesti, SuomiFiClient, Viesti } from "../../src/suomifi/viranomaispalvelutwsinterface/suomifi";
import { HaeTilaTietoResponse, ViranomaispalvelutWsInterfaceClient } from "../../src/suomifi/viranomaispalvelutwsinterface";
import { PublishOrExpireEventType } from "../../src/sqsEvents/projektiScheduleManager";
import { fileService } from "../../src/files/fileService";
import { DBProjekti } from "../../src/database/model";
import { DBOmistaja } from "../../src/mml/kiinteistoHandler";
import { fail } from "assert";

type SuomiFiRequest = {
  viesti?: Viesti;
  pdfViesti?: PdfViesti;
};

function mockSuomiFiClient(request: SuomiFiRequest, asiakasTila: number, pdfTilakoodi = 202): SuomiFiClient {
  return {
    getSoapClient: () => undefined as unknown as ViranomaispalvelutWsInterfaceClient,
    haeAsiakas: () => {
      return new Promise((resolve) => {
        resolve({
          HaeAsiakkaitaResult: {
            TilaKoodi: { TilaKoodi: 0 },
            Asiakkaat: { Asiakas: [{ Tila: asiakasTila, attributes: { AsiakasTunnus: "ABC", TunnusTyyppi: "SSN" } }] },
          },
        });
      });
    },
    rajapinnanTila: () => {
      return undefined as unknown as Promise<HaeTilaTietoResponse>;
    },
    lahetaInfoViesti(viesti) {
      request.viesti = viesti;
      return new Promise((resolve) => {
        resolve({
          LisaaKohteitaResult: {
            TilaKoodi: { TilaKoodi: 0 },
            Kohteet: { Kohde: [{ Asiakas: [{ KohteenTila: 200, attributes: { AsiakasTunnus: "ABC", TunnusTyyppi: "SSN" } }] }] },
          },
        });
      });
    },
    lahetaViesti(viesti) {
      request.pdfViesti = viesti;
      return new Promise((resolve) => {
        resolve({ LahetaViestiResult: { TilaKoodi: { TilaKoodi: pdfTilakoodi } } });
      });
    },
  };
}

describe("suomifiHandler", () => {
  after(() => {
    //setClient(undefined);
  });

  before(() => {
    //setClient(mockMmlClient);
  });

  beforeEach(() => {
    setLogContextOid("1");
    identifyMockUser({ etunimi: "", sukunimi: "", uid: "testuid", __typename: "NykyinenKayttaja" });
  });

  afterEach(() => {
    //parameterStub.restore();
  });

  it("muistuttajan sähköposti annettu", async () => {
    const parameterStub = sinon.stub(parameters, "isSuomiFiIntegrationEnabled").resolves(false);
    const muistuttaja: DBMuistuttaja = {
      id: "123",
      expires: 0,
      lisatty: now().toString(),
      oid: "1",
      sahkoposti: "test@test.fi",
      muistutus: "Muistutus 1",
      liite: "test.txt",
    };
    const emailStub = sinon
      .stub(emailClient, "sendEmail")
      .resolves({ accepted: [""], messageId: "", pending: [], rejected: [], response: "", envelope: { from: "", to: ["test@test.fi"] } });
    mockClient(DynamoDBDocumentClient)
      .on(GetCommand, { TableName: config.muistuttajaTableName })
      .resolves({ Item: muistuttaja })
      .on(GetCommand, { TableName: config.projektiTableName })
      .resolves({ Item: { id: "1", velho: { nimi: "Projektin nimi", asiatunnusVayla: "vayla123", asiatunnusELY: "ely123" } } });
    const body: SuomiFiSanoma = { muistuttajaId: "123" };
    const msg = { Records: [{ body: JSON.stringify(body) }] };
    await handleEvent(msg as SQSEvent);
    expect(emailStub.callCount).to.equal(1);
    expect(emailStub.args[0]).toMatchSnapshot();
    mockClient(DynamoDBDocumentClient)
      .on(GetCommand, { TableName: config.muistuttajaTableName })
      .resolves({ Item: muistuttaja })
      .on(GetCommand, { TableName: config.projektiTableName })
      .resolves({
        Item: {
          id: "1",
          velho: {
            nimi: "Projektin nimi2",
            suunnittelustaVastaavaViranomainen: SuunnittelustaVastaavaViranomainen.VAYLAVIRASTO,
            asiatunnusVayla: "vayla123",
            asiatunnusELY: "ely123",
          },
        },
      });
    await handleEvent(msg as SQSEvent);
    expect(emailStub.callCount).to.equal(2);
    expect(emailStub.args[1]).toMatchSnapshot();
    delete muistuttaja.liite;
    mockClient(DynamoDBDocumentClient)
      .on(GetCommand, { TableName: config.muistuttajaTableName })
      .resolves({ Item: muistuttaja })
      .on(GetCommand, { TableName: config.projektiTableName })
      .resolves({
        Item: {
          id: "1",
          velho: {
            nimi: "Projektin nimi3",
            suunnittelustaVastaavaViranomainen: SuunnittelustaVastaavaViranomainen.VAYLAVIRASTO,
            asiatunnusVayla: "vayla123",
            asiatunnusELY: "ely123",
          },
        },
      });
    await handleEvent(msg as SQSEvent);
    expect(emailStub.callCount).to.equal(3);
    expect(emailStub.args[2]).toMatchSnapshot();
    parameterStub.restore();
    emailStub.restore();
  });
  it("muistuttajan sähköpostia ei annettu", async () => {
    const parameterStub = sinon.stub(parameters, "isSuomiFiIntegrationEnabled").resolves(false);
    const muistuttaja: DBMuistuttaja = {
      id: "123",
      expires: 0,
      lisatty: now().toString(),
      oid: "1",
      sahkoposti: "",
      muistutus: "Muistutus 1",
      liite: "test.txt",
    };
    const emailStub = sinon
      .stub(emailClient, "sendEmail")
      .resolves({ accepted: [""], messageId: "", pending: [], rejected: [], response: "", envelope: { from: "", to: ["test@test.fi"] } });
    mockClient(DynamoDBDocumentClient)
      .on(GetCommand, { TableName: config.muistuttajaTableName })
      .resolves({ Item: muistuttaja })
      .on(GetCommand, { TableName: config.projektiTableName })
      .resolves({ Item: { id: "1", velho: { nimi: "Projektin nimi", asiatunnusVayla: "vayla123", asiatunnusELY: "ely123" } } });
    const body: SuomiFiSanoma = { muistuttajaId: "123" };
    const msg = { Records: [{ body: JSON.stringify(body) }] };
    await handleEvent(msg as SQSEvent);
    expect(emailStub.callCount).to.equal(0);
    parameterStub.restore();
    emailStub.restore();
  });
  it("muistuttajan viesti suomi.fi", async () => {
    const parameterStub = sinon.stub(parameters, "isSuomiFiIntegrationEnabled").resolves(true);
    const muistuttaja: DBMuistuttaja = {
      id: "123",
      expires: 0,
      lisatty: now().toString(),
      oid: "1",
      sahkoposti: "",
      henkilotunnus: "ABC",
      muistutus: "Muistutus 1",
      liite: "test.txt",
    };
    const request: SuomiFiRequest = {};
    const client = mockSuomiFiClient(request, 300);
    setMockSuomiFiClient(client);
    const mock = mockClient(DynamoDBDocumentClient)
      .on(GetCommand, { TableName: config.muistuttajaTableName })
      .resolves({ Item: muistuttaja })
      .on(GetCommand, { TableName: config.projektiTableName })
      .resolves({ Item: { id: "1", velho: { nimi: "Projektin nimi", asiatunnusVayla: "vayla123", asiatunnusELY: "ely123" } } });
    const body: SuomiFiSanoma = { muistuttajaId: "123" };
    const msg = { Records: [{ body: JSON.stringify(body) }] };
    await handleEvent(msg as SQSEvent);
    expect(request.viesti).toMatchSnapshot();
    expect(mock.commandCalls(UpdateCommand).length).to.equal(0);
    parameterStub.restore();
  });
  it("muistuttajan viesti suomi.fi paperiposti", async () => {
    const parameterStub = sinon.stub(parameters, "isSuomiFiIntegrationEnabled").resolves(true);
    const muistuttaja: DBMuistuttaja = {
      id: "123",
      expires: 0,
      lisatty: now().toString(),
      oid: "1",
      sahkoposti: "test@test.fi",
      henkilotunnus: "ABC",
      muistutus: "Muistutus 1",
      liite: "test.txt",
    };
    const request: SuomiFiRequest = {};
    const client = mockSuomiFiClient(request, 310);
    setMockSuomiFiClient(client);
    const mock = mockClient(DynamoDBDocumentClient)
      .on(GetCommand, { TableName: config.muistuttajaTableName })
      .resolves({ Item: muistuttaja })
      .on(GetCommand, { TableName: config.projektiTableName })
      .resolves({ Item: { id: "1", velho: { nimi: "Projektin nimi", asiatunnusVayla: "vayla123", asiatunnusELY: "ely123" } } });
    const emailStub = sinon
      .stub(emailClient, "sendEmail")
      .resolves({ accepted: [""], messageId: "", pending: [], rejected: [], response: "", envelope: { from: "", to: ["test@test.fi"] } });
    const body: SuomiFiSanoma = { muistuttajaId: "123" };
    const msg = { Records: [{ body: JSON.stringify(body) }] };
    await handleEvent(msg as SQSEvent);
    expect(request.viesti).to.equal(undefined);
    expect(mock.commandCalls(UpdateCommand).length).to.equal(0);
    expect(emailStub.callCount).to.equal(1);
    expect(emailStub.args[0]).toMatchSnapshot();
    parameterStub.restore();
    emailStub.restore();
  });
  it("muistuttajan pdf viesti suomi.fi", async () => {
    const parameterStub = sinon.stub(parameters, "isSuomiFiIntegrationEnabled").resolves(true);
    const muistuttaja: DBMuistuttaja = {
      id: "123",
      expires: 0,
      lisatty: now().toString(),
      oid: "1",
      sahkoposti: "",
      henkilotunnus: "ABC",
      muistutus: "Muistutus 1",
      liite: "test.txt",
      etunimi: "Testi",
      sukunimi: "Teppo",
      lahiosoite: "Osoite 1",
      postinumero: "00100",
      postitoimipaikka: "Helsinki",
    };
    const request: SuomiFiRequest = {};
    const client = mockSuomiFiClient(request, 300);
    setMockSuomiFiClient(client);
    const dbProjekti: Partial<DBProjekti> = {oid: "1", nahtavillaoloVaihe: { id: 1},
    velho: { nimi: "Projektin nimi", asiatunnusVayla: "vayla123", asiatunnusELY: "ely123"}};
    const mock = mockClient(DynamoDBDocumentClient)
      .on(GetCommand, { TableName: config.muistuttajaTableName })
      .resolves({ Item: muistuttaja })
      .on(GetCommand, { TableName: config.projektiTableName })
      .resolves({
        Item: dbProjekti,
      });
    const fileStub = sinon.stub(fileService, "getProjektiFile").resolves(Buffer.from("tiedosto"));
    const body: SuomiFiSanoma = { muistuttajaId: "123", tyyppi: PublishOrExpireEventType.PUBLISH_NAHTAVILLAOLO };
    const msg = { Records: [{ body: JSON.stringify(body) }] };
    await handleEvent(msg as SQSEvent);
    expect(request.pdfViesti).toMatchSnapshot();
    expect(mock.commandCalls(UpdateCommand).length).to.equal(1);
    expect(mock.commandCalls(UpdateCommand)[0].args[0].input).toMatchSnapshot();
    parameterStub.restore();
    fileStub.restore();
  });
  it("omistajan pdf viesti suomi.fi", async () => {
    const parameterStub = sinon.stub(parameters, "isSuomiFiIntegrationEnabled").resolves(true);
    const omistaja: DBOmistaja = {
      id: "123",
      expires: 0,
      lisatty: now().toString(),
      oid: "1",
      henkilotunnus: "ABC",
      etunimet: "Testi",
      sukunimi: "Teppo",
      jakeluosoite: "Osoite 1",
      postinumero: "00100",
      paikkakunta: "Helsinki",
      kiinteistotunnus: "123",
    };
    const request: SuomiFiRequest = {};
    const client = mockSuomiFiClient(request, 300);
    setMockSuomiFiClient(client);
    const dbProjekti: Partial<DBProjekti> = {oid: "1", nahtavillaoloVaihe: { id: 1},
    velho: { nimi: "Projektin nimi", asiatunnusVayla: "vayla123", asiatunnusELY: "ely123"}};
    const mock = mockClient(DynamoDBDocumentClient)
      .on(GetCommand, { TableName: config.omistajaTableName })
      .resolves({ Item: omistaja })
      .on(GetCommand, { TableName: config.projektiTableName })
      .resolves({
        Item: dbProjekti,
      });
    const fileStub = sinon.stub(fileService, "getProjektiFile").resolves(Buffer.from("tiedosto"));
    const body: SuomiFiSanoma = { omistajaId: "123", tyyppi: PublishOrExpireEventType.PUBLISH_NAHTAVILLAOLO };
    const msg = { Records: [{ body: JSON.stringify(body) }] };
    await handleEvent(msg as SQSEvent);
    expect(request.pdfViesti).toMatchSnapshot();
    expect(mock.commandCalls(UpdateCommand).length).to.equal(1);
    expect(mock.commandCalls(UpdateCommand)[0].args[0].input).toMatchSnapshot();
    parameterStub.restore();
    fileStub.restore();
  });
  it("omistajan pdf viesti suomi.fi epäonnistui", async () => {
    const parameterStub = sinon.stub(parameters, "isSuomiFiIntegrationEnabled").resolves(true);
    const omistaja: DBOmistaja = {
      id: "123",
      expires: 0,
      lisatty: now().toString(),
      oid: "1",
      henkilotunnus: "ABC",
      etunimet: "Testi",
      sukunimi: "Teppo",
      jakeluosoite: "Osoite 1",
      postinumero: "00100",
      paikkakunta: "Helsinki",
      kiinteistotunnus: "123",
    };
    const request: SuomiFiRequest = {};
    const client = mockSuomiFiClient(request, 300, 200);
    setMockSuomiFiClient(client);
    const dbProjekti: Partial<DBProjekti> = {oid: "1", nahtavillaoloVaihe: { id: 1},
    velho: { nimi: "Projektin nimi", asiatunnusVayla: "vayla123", asiatunnusELY: "ely123"}};
    const mock = mockClient(DynamoDBDocumentClient)
      .on(GetCommand, { TableName: config.omistajaTableName })
      .resolves({ Item: omistaja })
      .on(GetCommand, { TableName: config.projektiTableName })
      .resolves({
        Item: dbProjekti,
      });
    const fileStub = sinon.stub(fileService, "getProjektiFile").resolves(Buffer.from("tiedosto"));
    const body: SuomiFiSanoma = { omistajaId: "123", tyyppi: PublishOrExpireEventType.PUBLISH_NAHTAVILLAOLO };
    const msg = { Records: [{ body: JSON.stringify(body) }] };
    try {
      await handleEvent(msg as SQSEvent);
      fail("Pitäisi heittää poikkeus");
    } catch(e) {
      assert(e instanceof Error);
      expect(e.message).to.equal("Suomi.fi pdf-viestin lähetys epäonnistui");
    }
    expect(mock.commandCalls(UpdateCommand).length).to.equal(1);
    expect(mock.commandCalls(UpdateCommand)[0].args[0].input).toMatchSnapshot();
    parameterStub.restore();
    fileStub.restore();
  });
});
