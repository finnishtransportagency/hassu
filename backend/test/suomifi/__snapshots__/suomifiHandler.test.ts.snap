// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`suomifiHandler muistuttajan pdf viesti suomi.fi 1`] = `
Object {
  "hetu": "ABC",
  "kustannuspaikka": undefined,
  "lahiosoite": "Osoite 1",
  "maa": "FI",
  "nimi": "Testi Teppo",
  "otsikko": "Ilmoitus suunnitelman nähtäville asettamisesta",
  "postinumero": "00100",
  "postitoimipaikka": "Helsinki",
  "sisalto": "Hei,

Olette saaneet kirjeen, jossa kerrotaan suunnitelman nähtäville asettamista sekä mahdollisuudesta tehdä suunnitelmasta muistutus. Kirje on tämän viestin liitteenä. Löydät kirjeestä linkin Valtion liikenneväylien suunnittelu -palveluun, missä pääsette tutustumaan suunnitelmaan tarkemmin.

Ystävällisin terveisin
Väylävirasto",
  "suunnittelustaVastaavaViranomainen": "VAYLAVIRASTO",
  "tiedosto": Object {
    "kuvaus": "T415 Ilmoitus kiinteistönomistajille suunnitelman nähtävilläolo",
    "nimi": "T415 Ilmoitus kiinteistonomistajille suunnitelman nahtavillaolo.pdf",
    "sisalto": Object {
      "data": Array [],
      "type": "Buffer",
    },
  },
  "ytunnus": undefined,
}
`;

exports[`suomifiHandler muistuttajan pdf viesti suomi.fi ruotsi 1`] = `
Object {
  "hetu": "ABC",
  "kustannuspaikka": undefined,
  "lahiosoite": "Svart katten's gatan 13 c 13, B.O.X 1010",
  "maa": "SE",
  "nimi": "Minerva Marttila",
  "otsikko": "Ilmoitus suunnitelman nähtäville asettamisesta",
  "postinumero": "SE-",
  "postitoimipaikka": "01230 STOCKHOLM, Ruotsii",
  "sisalto": "Hei,

Olette saaneet kirjeen, jossa kerrotaan suunnitelman nähtäville asettamista sekä mahdollisuudesta tehdä suunnitelmasta muistutus. Kirje on tämän viestin liitteenä. Löydät kirjeestä linkin Valtion liikenneväylien suunnittelu -palveluun, missä pääsette tutustumaan suunnitelmaan tarkemmin.

Ystävällisin terveisin
Väylävirasto",
  "suunnittelustaVastaavaViranomainen": "VAYLAVIRASTO",
  "tiedosto": Object {
    "kuvaus": "T415 Ilmoitus kiinteistönomistajille suunnitelman nähtävilläolo",
    "nimi": "T415 Ilmoitus kiinteistonomistajille suunnitelman nahtavillaolo.pdf",
    "sisalto": Object {
      "data": Array [],
      "type": "Buffer",
    },
  },
  "ytunnus": undefined,
}
`;

exports[`suomifiHandler muistuttajan sähköposti annettu 1`] = `
Array [
  Object {
    "subject": "Muistutus on vastaanotettu",
    "text": "Muistutus on vastaanotettu
Suunnitelman nimi: Projektin nimi
Suunnitelman asiatunnus: ely123

ELY-keskus käsittelee muistutustasi asiatunnuksella. Mikäli haluat olla yhteydessä suunnitelmaan liittyen ilmoitathan asiatunnuksen viestissäsi.

Suunnitelman tietoihin pääset tästä linkistä: https://localhost:3000/suunnitelma/1/nahtavillaolo

Muistutus:
Muistutus 1

Muistutuksen mukana toimitettu seuraavannimiset liitteet:
test.txt
",
    "to": "test@test.fi",
  },
]
`;

exports[`suomifiHandler muistuttajan sähköposti annettu 2`] = `
Array [
  Object {
    "subject": "Muistutus on vastaanotettu",
    "text": "Muistutus on vastaanotettu
Suunnitelman nimi: Projektin nimi2
Suunnitelman asiatunnus: vayla123

Väylävirasto käsittelee muistutustasi asiatunnuksella. Mikäli haluat olla yhteydessä suunnitelmaan liittyen ilmoitathan asiatunnuksen viestissäsi.

Suunnitelman tietoihin pääset tästä linkistä: https://localhost:3000/suunnitelma/1/nahtavillaolo

Muistutus:
Muistutus 1

Muistutuksen mukana toimitettu seuraavannimiset liitteet:
test.txt
",
    "to": "test@test.fi",
  },
]
`;

exports[`suomifiHandler muistuttajan sähköposti annettu 3`] = `
Array [
  Object {
    "subject": "Muistutus on vastaanotettu",
    "text": "Muistutus on vastaanotettu
Suunnitelman nimi: Projektin nimi3
Suunnitelman asiatunnus: vayla123

Väylävirasto käsittelee muistutustasi asiatunnuksella. Mikäli haluat olla yhteydessä suunnitelmaan liittyen ilmoitathan asiatunnuksen viestissäsi.

Suunnitelman tietoihin pääset tästä linkistä: https://localhost:3000/suunnitelma/1/nahtavillaolo

Muistutus:
Muistutus 1


",
    "to": "test@test.fi",
  },
]
`;

exports[`suomifiHandler muistuttajan sähköposti annettu 4`] = `
Array [
  Object {
    "subject": "Muistutus on vastaanotettu / Anmärkningen har mottagits",
    "text": "Muistutus on vastaanotettu
Suunnitelman nimi: Projektin nimi4
Suunnitelman asiatunnus: vayla123

Väylävirasto käsittelee muistutustasi asiatunnuksella. Mikäli haluat olla yhteydessä suunnitelmaan liittyen ilmoitathan asiatunnuksen viestissäsi.

Suunnitelman tietoihin pääset tästä linkistä: https://localhost:3000/suunnitelma/1/nahtavillaolo

Muistutus:
Muistutus 1



Anmärkningen har mottagits
Planens namn: Projekts namn4
Planens ärendekod: vayla123

Trafikledsverket behandlar din anmärkning med en ärendekod. Om du vill kontakta oss i anslutning till planen, meddela ärendekoden i ditt meddelande.

Du kommer till planens uppgifter via denna länk: https://localhost:3000/sv/suunnitelma/1/nahtavillaolo

Anmärkning:
Muistutus 1


",
    "to": "test@test.fi",
  },
]
`;

exports[`suomifiHandler muistuttajan sähköposti annettu 5`] = `
Array [
  Object {
    "subject": "Muistutus on vastaanotettu / Anmärkningen har mottagits",
    "text": "Muistutus on vastaanotettu
Suunnitelman nimi: Projektin nimi5
Suunnitelman asiatunnus: ely123

ELY-keskus käsittelee muistutustasi asiatunnuksella. Mikäli haluat olla yhteydessä suunnitelmaan liittyen ilmoitathan asiatunnuksen viestissäsi.

Suunnitelman tietoihin pääset tästä linkistä: https://localhost:3000/suunnitelma/1/nahtavillaolo

Muistutus:
Muistutus 1

Muistutuksen mukana toimitettu seuraavannimiset liitteet:
test.txt

Anmärkningen har mottagits
Planens namn: Projekts namn5
Planens ärendekod: ely123

NTM-centralen behandlar din anmärkning med en ärendekod. Om du vill kontakta oss i anslutning till planen, meddela ärendekoden i ditt meddelande.

Du kommer till planens uppgifter via denna länk: https://localhost:3000/sv/suunnitelma/1/nahtavillaolo

Anmärkning:
Muistutus 1

Bilagor med följande namn som bifogats anmärkningen:
test.txt
",
    "to": "test@test.fi",
  },
]
`;

exports[`suomifiHandler muistuttajan viesti suomi.fi 1`] = `
Object {
  "hetu": "ABC",
  "otsikko": "Muistutus on vastaanotettu",
  "sisalto": "Muistutus on vastaanotettu
Suunnitelman nimi: Projektin nimi
Suunnitelman asiatunnus: ely123

ELY-keskus käsittelee muistutustasi asiatunnuksella. Mikäli haluat olla yhteydessä suunnitelmaan liittyen ilmoitathan asiatunnuksen viestissäsi.

Suunnitelman tietoihin pääset tästä linkistä: https://localhost:3000/suunnitelma/1/nahtavillaolo

Muistutus:
Muistutus 1

Muistutuksen mukana toimitettu seuraavannimiset liitteet:
test.txt
",
}
`;

exports[`suomifiHandler muistuttajan viesti suomi.fi paperiposti 1`] = `
Array [
  Object {
    "subject": "Muistutus on vastaanotettu",
    "text": "Muistutus on vastaanotettu
Suunnitelman nimi: Projektin nimi
Suunnitelman asiatunnus: ely123

ELY-keskus käsittelee muistutustasi asiatunnuksella. Mikäli haluat olla yhteydessä suunnitelmaan liittyen ilmoitathan asiatunnuksen viestissäsi.

Suunnitelman tietoihin pääset tästä linkistä: https://localhost:3000/suunnitelma/1/nahtavillaolo

Muistutus:
Muistutus 1

Muistutuksen mukana toimitettu seuraavannimiset liitteet:
test.txt
",
    "to": "test@test.fi",
  },
]
`;

exports[`suomifiHandler omistajan pdf viesti suomi.fi 1`] = `
Object {
  "hetu": "ABC",
  "kustannuspaikka": undefined,
  "lahiosoite": "Osoite 1",
  "maa": "FI",
  "nimi": "Testi Teppo",
  "otsikko": "Ilmoitus suunnitelman nähtäville asettamisesta",
  "postinumero": "00100",
  "postitoimipaikka": "Helsinki",
  "sisalto": "Hei,

Olette saaneet kirjeen, jossa kerrotaan suunnitelman nähtäville asettamista sekä mahdollisuudesta tehdä suunnitelmasta muistutus. Kirje on tämän viestin liitteenä. Löydät kirjeestä linkin Valtion liikenneväylien suunnittelu -palveluun, missä pääsette tutustumaan suunnitelmaan tarkemmin.

Ystävällisin terveisin
Väylävirasto",
  "suunnittelustaVastaavaViranomainen": "VAYLAVIRASTO",
  "tiedosto": Object {
    "kuvaus": "T415 Ilmoitus kiinteistönomistajille suunnitelman nähtävilläolo",
    "nimi": "T415 Ilmoitus kiinteistonomistajille suunnitelman nahtavillaolo.pdf",
    "sisalto": Object {
      "data": Array [],
      "type": "Buffer",
    },
  },
  "ytunnus": undefined,
}
`;

exports[`suomifiHandler omistajan pdf viesti suomi.fi hyväksymispäätös 1`] = `
Object {
  "hetu": "ABC",
  "kustannuspaikka": undefined,
  "lahiosoite": "Osoite 1",
  "maa": "FI",
  "nimi": "Testi Teppo",
  "otsikko": "Ilmoitus suunnitelman hyväksymispäätöksestä / Meddelande om beslut om godkännande av planen",
  "postinumero": "00100",
  "postitoimipaikka": "Helsinki",
  "sisalto": "Hei,

Olette saaneet kirjeen, jossa kerrotaan suunnitelmaa koskevasta hyväksymispäätöksestä. Kirje on tämän viestin liitteenä. Löydät kirjeestä linkin Valtion liikenneväylien suunnittelu -palveluun, missä pääsette tutustumaan tarkemmin päätökseen ja sen liitteenä oleviin suunnitelma-aineistoihin.

Ystävällisin terveisin
Pirkanmaan ELY-keskus

Hej,

Ni har fått ett brev med information om beslut om godkännande av planen. Brevet finns som bilaga till detta meddelande. I brevet hittar du en länk till tjänsten Planering av statens trafikleder, där ni kan bekanta er närmare med beslutet och det bifogade planeringsmaterialet.

Med vänlig hälsning
NTM-centralen i Birkaland
",
  "suunnittelustaVastaavaViranomainen": "PIRKANMAAN_ELY",
  "tiedosto": Object {
    "kuvaus": "T431_4 Ilmoitus hyväksymispäätöksestä kiinteistönomistajille ja muistuttajille",
    "nimi": "T431_4 Ilmoitus hyvaksymispaatoksesta kiinteistonomistajille ja muistuttajille.pdf",
    "sisalto": Object {
      "data": Array [],
      "type": "Buffer",
    },
  },
  "ytunnus": undefined,
}
`;

exports[`suomifiHandler omistajan pdf viesti suomi.fi nähtävilläolo 1`] = `
Object {
  "hetu": "ABC",
  "kustannuspaikka": undefined,
  "lahiosoite": "Osoite 1",
  "maa": "FI",
  "nimi": "Testi Teppo",
  "otsikko": "Ilmoitus suunnitelman nähtäville asettamisesta / Meddelande om framläggande av planen",
  "postinumero": "00100",
  "postitoimipaikka": "Helsinki",
  "sisalto": "Hei,

Olette saaneet kirjeen, jossa kerrotaan suunnitelman nähtäville asettamista sekä mahdollisuudesta tehdä suunnitelmasta muistutus. Kirje on tämän viestin liitteenä. Löydät kirjeestä linkin Valtion liikenneväylien suunnittelu -palveluun, missä pääsette tutustumaan suunnitelmaan tarkemmin.

Ystävällisin terveisin
Uudenmaan ELY-keskus

Hej,

Ni har fått ett brev med information om framläggandet av planen samt om möjligheten att göra en anmärkning på planen. Brevet finns som bilaga till detta meddelande. I brevet hittar du en länk till tjänsten Planering av statens trafikleder, där ni kan bekanta er närmare med planen.

Med vänlig hälsning
NTM-centralen i Nyland
",
  "suunnittelustaVastaavaViranomainen": "UUDENMAAN_ELY",
  "tiedosto": Object {
    "kuvaus": "T415 Ilmoitus kiinteistönomistajille suunnitelman nähtävilläolo",
    "nimi": "T415 Ilmoitus kiinteistonomistajille suunnitelman nahtavillaolo.pdf",
    "sisalto": Object {
      "data": Array [],
      "type": "Buffer",
    },
  },
  "ytunnus": undefined,
}
`;

exports[`suomifiHandler omistajan suomi-saame pdf viesti suomi.fi nähtävilläolo 1`] = `
Object {
  "hetu": "ABC",
  "kustannuspaikka": undefined,
  "lahiosoite": "Osoite 1",
  "maa": "FI",
  "nimi": "Testi Teppo",
  "otsikko": "Ilmoitus suunnitelman nähtäville asettamisesta",
  "postinumero": "00100",
  "postitoimipaikka": "Helsinki",
  "sisalto": "Hei,

Olette saaneet kirjeen, jossa kerrotaan suunnitelman nähtäville asettamista sekä mahdollisuudesta tehdä suunnitelmasta muistutus. Kirje on tämän viestin liitteenä. Löydät kirjeestä linkin Valtion liikenneväylien suunnittelu -palveluun, missä pääsette tutustumaan suunnitelmaan tarkemmin.

Ystävällisin terveisin
Uudenmaan ELY-keskus",
  "suunnittelustaVastaavaViranomainen": "UUDENMAAN_ELY",
  "tiedosto": Object {
    "kuvaus": "T415 Ilmoitus kiinteistönomistajille suunnitelman nähtävilläolo",
    "nimi": "T415 Ilmoitus kiinteistonomistajille suunnitelman nahtavillaolo.pdf",
    "sisalto": Object {
      "data": Array [],
      "type": "Buffer",
    },
  },
  "ytunnus": undefined,
}
`;
